go/src/koding/klientctl/auth/register.go

```diff
@@ -0,0 +1,326 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"koding/httputil"
+	"koding/klientctl/config"
+	"koding/klientctl/ctlcli"
+	"koding/klientctl/endpoint/kloud"
+	"koding/klientctl/helper"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/codegangsta/cli"
+	"github.com/koding/logging"
+)
+
+// NewRegisterSubCommand provides the subcommand for registering
+func NewRegisterSubCommand(log logging.Logger) cli.Command {
+	return cli.Command{
+		Name:   "register",
+		Usage:  "Registers the user",
+		Action: ctlcli.ExitAction(RegisterCommand, log, "register"),
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "username, u",
+				Usage: "Username to register",
+			},
+			cli.StringFlag{
+				Name:  "firstName",
+				Usage: "firstName to register",
+			},
+			cli.StringFlag{
+				Name:  "lastName",
+				Usage: "lastName to register",
+			},
+			cli.StringFlag{
+				Name:  "password",
+				Usage: "Username password to register",
+			},
+			cli.StringFlag{
+				Name:  "email",
+				Usage: "Email Address to register",
+			},
+			cli.StringFlag{
+				Name:  "team",
+				Usage: "Team Name to register",
+			},
+			cli.StringFlag{
+				Name:  "company",
+				Usage: "Team Company Name to register (default: team name)",
+			},
+			cli.BoolFlag{
+				Name:  "newsletter",
+				Usage: "Do you want to get occasional newsletters from Koding?",
+			},
+			cli.BoolFlag{
```

Comment:
if you use strings to carry these vars, shouldn't this be a string too? because if user specifies --alreadyMember=false it will be asked of this anyway (` r.AlreadyMember == falseValue ` in  `checkAndAskRequiredFields` method)